### ISSUE TITLE
[Bug]:  Keyboard focus is not visible While invoking the “Tests completed” button. 

### ISSUE THREAD

**Issue by msftedad** [storybookjs/storybook#24129]
### Describe the bug

Keyboard focus is not visible While invoking the “Tests completed” button. 

Actual Result:
Keyboard focus is not visible While invoking the “Tests completed” button. 

Expected Result:
Keyboard focus should be visible While invoking the Tests completed” button.

### To Reproduce

1. Open URL: [Common Navigation - Scrolling Panel ⋅ Storybook (windows.net)](https://vivacommonstorybooktest.blob.core.windows.net/$web/index.html?path=/story/common-navigation--scrolling-panel&globals=backgrounds.grid:!false;backgrounds.value:transparent)	
2. Navigate to “App bar” button at right side of the pane invoke it and navigate to “Default” link invoke it. 
3. Now navigate to "Accessibility" tab item invoke it and navigate "Tests completed" button invoke it observe the keyboard focus 

Attachments:
[keyboard focus not visible tests completed.webm](https://github.com/storybookjs/storybook/assets/93735775/1a5e2859-4af3-4c75-a471-b287a2689b1b)

### System

```bash
OS: Windows 11
OS version: 22H2 (OS Build 22621.2134)
Edge Version: 116.0.1938.69 (Official build) (64-bit)
```

**Comment by Nevnet99** [comment]
I can take a look at this 😊

**Comment by Nevnet99** [comment]
Having some trouble building up a sandbox with my local code for the a11y package can anyone point me in the right direction ? @shilman

**Comment by abhisheksharma010** [comment]
Hey there! 👋

I'm new to open source and super excited to dive in! I'd love to work on this. Can I be assigned to this issue? 🚀

**Comment by Nevnet99** [comment]
all good got some help from the discord ! will be commit a PR soon :)

**Comment by Nevnet99** [comment]
PR is here ! :) https://github.com/storybookjs/storybook/pull/24446

**Comment by yannbf** [comment]
Hey @msftedad, thanks for reporting this issue! I believe it's incorrect to say that the keyboard focus is not visible. What happens is that you have focus on the element, which shows the outline, and once you press enter, the tests re-run and the panel re-renders. The focus is then _lost_, moved up to the top of the panel.

I am not sure if that is a bad thing, because after you run the tests, you will likely want to go through the results, and not run the tests again. If you were to run the tests again, you probably would do so after making changes or interacting with the component, which will make you lose keyboard focus anyway.

What do you think is the right approach here?

**Comment by msftedad** [comment]
Hi @yannbf, Thanks for the information. We will connect with the PO team and will get back to you soon.

**Comment by yannbf** [comment]
Thank you @msftedad, eager to hear back!

**Comment by Sidnioulz** [comment]
@yannbf I think it may be considered a bug. If focus is lost, the question for low/no vision users who use a screenreader is: where the heck in the application did I land?

We could consider working around this by having a live announcement that says "Tests are now running, focus has been moved to the sidebar". This could make sense as we know we already want to create ARIA live notifications to announce test results.

It may also be possible and relatively easy to preserve the tabbar with the button, so that kb focus isn't lost.

**Comment by Sidnioulz** [comment]
We want to refactor the UI so that the overall bar with tools is preserved. This means creating the AriaTabs without tabs and with `showToolsWhenEmpty`, and letting the block of content for `status==="running"` show under it.

**Comment by ghengeveld** [comment]
Closing as this is no longer relevant with the latest version of the Accessibility addon.

**Comment by Sidnioulz** [comment]
> Closing as this is no longer relevant with the latest version of the Accessibility addon.

For what it's worth, the new UI has "rerun tests" buttons that have the same bug. Should we adjust the scope of the issue rather than close it?
